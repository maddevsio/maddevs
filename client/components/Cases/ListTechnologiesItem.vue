<template>
  <li class="case_technologies-item">
    <div class="case_technologies-item__tech-icon" :class="className"></div>
    <p class="case_technologies-item__tech-name">{{techName}}</p>
  </li>
</template>

<script>
export default {
  name: 'ListTechnologiesItem',
  props: {
    techName: {
      type: String,
      default: ''
    },
    className: {
      type: String,
      default: ''
    }
  }
};
</script>

<style scoped lang="scss">
  @import '../../assets/styles/cases/_mixins';
  @import '../../assets/styles/_techIcons';

  .case_technologies-item {
    display: flex;
    flex-direction: column;
    align-items: center;

    &__tech-icon {
      margin-bottom: 5px;

      &::before {
        content: '';
        width: 40px;
        height: 40px;
        display: block;
      }
    }

    &__tech-name {
      @include default_text($text-color--tech-label-black, 14px, 171%, -0.035em, normal);
    }

    &__docker {
      &::before {
        width: 70px;
      }
    }

    &__mobile-ios-android {
      &::before {
        width: 84px;
      }
    }

    @media screen and (max-width: 768px) {
      &__tech-icon {
        margin-bottom: 8px;

        &::before {
          width: 20px;
          height: 20px;
        }
      }

      &__docker {
        &::before {
          width: 50px;
        }
      }

      &__mobile-ios-android {
        &::before {
          width: 64px;
        }
      }

      &__tech-name {
        @include default_text($text-color--tech-label-black, 8px, 163%, -0.02em, normal);
        width: 40px;
        display: block;
        overflow: hidden;
        white-space: nowrap;
        text-overflow: ellipsis;
        text-align: center;
      }
    }
  }
</style>
