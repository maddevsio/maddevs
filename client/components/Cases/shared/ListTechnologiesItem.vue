<template>
  <li class="case_technologies-item">
    <img
      v-if="name"
      :data-src="require(`@/assets/img/Home/svg/technologies/${name}.svg`)"
      :alt="alt"
      class="img_lazy case_technologies-item__tech-icon"
      :class="`case_technologies-item__${name} ${name}`"
    >
    <p
      v-if="alt"
      class="case_technologies-item__tech-name"
    >
      {{ alt }}
    </p>
  </li>
</template>

<script>
export default {
  name: 'ListTechnologiesItem',
  props: {
    name: {
      type: String,
      default: '',
    },

    alt: {
      type: String,
      default: '',
    },
  },
}
</script>

<style scoped lang="scss">
@import '../../../assets/styles/cases/_mixins';

.case_technologies-item {
  display: flex;
  flex-direction: column;
  align-items: center;

  &__tech-icon {
    margin-bottom: 5px;
    width: 40px;
    height: 40px;
    display: block;
  }

  &__tech-name {
    @include default_text($text-color--tech-label-black, 14px, 171%, -0.035em, normal);
  }

  &__docker,
  &__grpc,
  &__backend-django {
    width: 70px;
  }

  &__uptime-robot,
  &__jquery,
  &__nimble,
  &__video-js,
  &__nginx-txt {
    width: 92px;
  }

  &__plex {
    width: 59px;
  }

  @media screen and (max-width: 768px) {
    &__tech-icon {
      margin-bottom: 8px;
      width: 20px;
      height: 20px;
    }

    &__docker,
    &__grpc,
    &__django,
    &__nginx-txt,
    &__video-js {
      width: 40px;
    }

    &__jquery,
    &__nimble {
      width: 45px;
    }

    &__uptime-robot {
      width: 62px;
    }

    &__plex {
      width: 28px;
    }

    &__tech-name {
      @include default_text($text-color--tech-label-black, 8px, 163%, -0.02em, normal);
      width: 40px;
      display: block;
      overflow: hidden;
      white-space: nowrap;
      text-overflow: ellipsis;
      text-align: center;
    }
  }
}
</style>
