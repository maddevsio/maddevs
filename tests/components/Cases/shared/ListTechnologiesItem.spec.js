import ListTechnologiesItem from '@/components/Cases/shared/ListTechnologiesItem'
import { render } from '@testing-library/vue'

describe('ListTechnologiesItem components', () => {
  it('should render correctly', () => {
    const { container } = render(ListTechnologiesItem)

    expect(container).toMatchSnapshot()
  })
})
