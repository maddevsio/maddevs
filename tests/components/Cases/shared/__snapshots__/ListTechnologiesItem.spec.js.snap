// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListTechnologiesItem components should render correctly 1`] = `
<div>
  <li
    class="case_technologies-item"
  >
    <!---->
     
    <!---->
  </li>
</div>
`;
